ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouterMetrics.java

```diff
@@ -267,6 +267,7 @@ public class NonBlockingRouterMetrics {
   private final Map<DataNodeId, NodeLevelMetrics> dataNodeToMetrics;
   private final RouterConfig routerConfig;
   private ScheduledExecutorService scheduler = null;
+  private CompressionMetrics compressionMetrics;
```

Comment:
we can just instantiate this object here, doing it in the method below might suffer some concurrency issue.

Reply:
Yes, we can do that.